### PR TITLE
Handle selective acks

### PR DESCRIPTION
**Pull request by KonradStaniec**
Adds handling of selective acks:
- generating bitmask required to selective ack packets
- handling of ack packet with selective ack bitmask
- responding to out of order packets with selective acks packet.

### CODE REVIEW

File: eth/utp/utp_socket.nim [hunk 1]
```diff
@@ -1014,6 +1121,7 @@ proc processPacket*(socket: UtpSocket, p: Packet) {.async.} =
 
       # we got in order packet
       if (pastExpected == 0 and (not socket.reachedFin)):
+        notice "Got in order packet"
```
Comment (kdeme) [inline eth/utp/utp_socket.nim:1124]: We'll have to revise all logs once as some are too chatty. This one too rather seems a debug log.

File: eth/utp/utp_socket.nim [hunk 2]
```diff
@@ -881,14 +861,129 @@ proc isAckNrInvalid(socket: UtpSocket, packet: Packet): bool =
     )
   )
 
+# counts the number of bytes acked by selective ack header
+proc calculateSelectiveAckBytes*(socket: UtpSocket,  receivedPackedAckNr: uint16, ext: SelectiveAckExtension): uint32 =
+  # we add 2, as the first bit in the mask therefore represents ackNr + 2 becouse
+  # ackNr + 1 (i.e next expected packet) is considered lost.
```
Comment (kdeme) [inline eth/utp/utp_socket.nim:867]:
```suggestion
  # Start from ack nr 2: the first bit in the mask represents `ackNr + 2` as
  # `ackNr + 1` (i.e next expected ack packet) is considered lost in this scenario.
```

File: eth/utp/utp_socket.nim [hunk 3]
```diff
@@ -881,14 +861,129 @@ proc isAckNrInvalid(socket: UtpSocket, packet: Packet): bool =
...
+      ackedBytes = ackedBytes + pkt.payloadLength
+    
+    dec bits
+
+  return ackedBytes
+
+# ack packets (removes them from out going buffer) based on selective ack extension header
+proc selectiveAckPackets(socket: UtpSocket,  receivedPackedAckNr: uint16, ext: SelectiveAckExtension, currentTime: Moment): void =
+   # we add 2, as the first bit in the mask therefore represents ackNr + 2 becouse
+  # ackNr + 1 (i.e next expected packet) is considered lost.
```
Comment (kdeme) [inline eth/utp/utp_socket.nim:902]:
```suggestion
  # Start from ack nr 2: the first bit in the mask represents `ackNr + 2` as
  # `ackNr + 1` (i.e next expected ack packet) is considered lost in this scenario.
```

File: eth/utp/utp_socket.nim (hunk 1 above)
Comment (KonradStaniec) [reply eth/utp/utp_socket.nim:1124]: Yep, I got in mind separeate task to change most logs into correct debug or info levels.

Comment (KonradStaniec) [reply eth/utp/utp_socket.nim:1124]: fix'ed this one here as this would be most chatty log.

**Pull request merged**